akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -185,6 +187,32 @@ object Source {
     new Source(scaladsl.Source.actorRef(bufferSize, overflowStrategy))
 
   /**
+   * Creates a synchronous (Java 6 compatible) Source from a Files contents.
+   * Emitted elements are [[ByteString]] of default 2048 bytes size.
+   *
+   * This source is backed by an Actor which will use the dedicated thread-pool base dispatcher.
+   * You can configure the default dispatcher for this Source by changing the `akka.stream.file-io-dispatcher` or
+   * set it for a given Source by using [[ActorOperationAttributes]].
+   *
+   * It materializes a [[Future]] containing the number of bytes read from the source file upon completion.
+   */
+  def synchronousFile(f: File): Source[ByteString, Future[Long]] =
```

Comment:
I don't think these API methods should be on Source. I would rather create a new object in the io subpackage. We shouldn't cram everything here.